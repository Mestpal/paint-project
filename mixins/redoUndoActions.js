import { mapActions } from 'vuex'

export default {
  computed: {
    // ...mapGetters('colorManager', [
    //   'selectedColor',
    //   'swatcherColors'
    // ])
  },
  methods: {
    ...mapActions('drawHistoryManager', [
      'undoLastDrawStep',
      'redoLastDrawStep',
      'saveDrawStepsHistory'
    ]),
    undoDrawStepAction () {
      this.undoLastDrawStep()
    },
    redoDrawStepAction () {
      this.redoLastDrawStep()
      console.log('redoDrawStepAction')
    },
    saveDrawHistoy () {
      this.saveDrawStepsHistory()
    }
  }
}
