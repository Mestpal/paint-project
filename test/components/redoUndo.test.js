
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import redoUndo from '@/components/redoUndone'
import { getters } from '@/mixins/drawModules/redoUndoActions'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('redoUndo.vue', () => {
  let store
  let wrapper

  beforeEach(() => {
    store = new Vuex.Store({
      getters
    })
    wrapper = shallowMount(redoUndo, {
      store,
      localVue
    })
  })

  it('redoUndo component renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
