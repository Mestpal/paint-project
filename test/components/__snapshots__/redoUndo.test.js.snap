// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redoUndo.vue redoUndo component renders correctly 1`] = `
<v-card-stub loaderheight="4" tag="div">
  <v-card-title-stub> Undo/Redo last action </v-card-title-stub>
  <v-card-text-stub>
    <v-row-stub tag="div" dense="true">
      <v-col-stub cols="3" tag="div">
        <v-btn-stub tag="button" activeclass="" icon="true" type="button">
          <v-icon-stub large="">
            mdi-undo
          </v-icon-stub>
        </v-btn-stub>
      </v-col-stub>
      <v-col-stub cols="3" tag="div">
        <v-btn-stub tag="button" activeclass="" icon="true" type="button">
          <v-icon-stub large="">
            mdi-redo
          </v-icon-stub>
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
  </v-card-text-stub>
</v-card-stub>
`;
